### DISCUSSION TITLE
Performance of the emulator

### DISCUSSION THREAD

**Discussion by anirbandan** [ReactiveCircus/android-emulator-runner#183] [General]
Hello,

I am using android-emulator-runner with appium for one of my test automation projects. I use a pixel 4a device to develop my test cases.

But to execute the test cases daily, I am using github actions.
env: macos-latest
on using the following config:
api-level: [29], arch: [x86_64], target: [default] with profile: Nexus 6
I get a very old emulator where everything looks zoomed in, app works slowly and crashes many a times.

I tried using another configuration for the emulator:
api-level: [29], arch: [x86_64], target: [google_apis] without any profile
Here, the recording shows the emulator is new (probably some pixel device) but the performance of the app gets poor and the app crashes within 20 seconds.

Note: the issue is not related to the app, as it works flawlessly on my local emulators and devices.

I am not expecting any physical device like performance but just want some suggestions on which configuration should I use to get the best performing emulator?

**Comment by arekkubaczkowski** [comment]
Hi, I am experiencing exactly the same, I would be grateful for any advices regarding this.

**Comment by ychescale9** [comment]
`profile: Galaxy Nexus` is usually faster. Older API levels are definitely faster. Unfortunately the performance and stability of the emulators on CI have never been great and there's nothing this action can do to reliably help.
